src/event/event.service.ts

```diff
@@ -1,4 +1,37 @@
 import { Injectable } from '@nestjs/common';
+import { PrismaService } from '../prisma/prisma.service';
 
 @Injectable()
-export class EventService {}
+export class EventService {
+  constructor(private prisma: PrismaService) {}
+
+  async findAllEvents(page: number, limit: number) {
+    const skip = (page - 1) * limit;
+    const totalEvents = await this.prisma.event.count({
+      where: {
+        deleted_at: null,
+      },
+    });
+
+    const events = await this.prisma.event.findMany({
+      skip: skip,
+      take: limit,
+      where: {
+        deleted_at: null,
+      },
+      include: {
+        location: true,
+      },
+      orderBy: {
+        start_time: 'asc',
+      },
+    });
+
```

Comment:
The current query only orders events by the earliest `start_time`, but it does not take the current date into account. To ensure we're only retrieving events that haven't passed, please update the query to filter out events with a start_time earlier than the current date.